### ISSUE TITLE
Master branch build fails

### ISSUE THREAD

**Issue by moritzschaefer** [sgratzl/lineupengine#13]
lineupjsx fails to build due to lineupengine. So I tried to compile lineupengine. There are a bunch of errors compiling it and I feel like I'm doing something wrong.

* Release number or git hash: master branch
 * tsc: Version 2.8.3
* tslint: 5.10.0

### Steps to reproduce

 clone, npm install, npm run build as documented in the README

### Observed behavior

```
ERROR in ./src/style/GridStyleManager.ts
Module parse failed: Unexpected token (14:32)
You may need an appropriate loader to handle this file type.
|
|
| export function setTemplate(root: HTMLElement) {
|   root.innerHTML = TEMPLATE;
|   return root;
 @ ./src/style/index.ts 2:0-97 2:0-97 2:0-97 2:0-97 2:0-97
 @ ./src/index.ts

ERROR in ./src/abortAble.ts
Module parse failed: Unexpected token (1:7)
You may need an appropriate loader to handle this file type.
| export interface IAbortAblePromise<T> extends Promise<T | symbol> {
|   abort(): void;
| }
 @ ./src/index.ts 4:0-28 4:0-28

ERROR in ./src/style/IColumn.ts
Module parse failed: Unexpected token (4:7)
You may need an appropriate loader to handle this file type.
|  * column base interface
|  */
| export interface IColumn {
|   readonly index: number;
|   readonly id: string;
 @ ./src/style/index.ts 1:0-34 1:0-34
 @ ./src/index.ts

ERROR in ./src/logic.ts
Module parse failed: Unexpected token (4:7)
You may need an appropriate loader to handle this file type.
|  * generic exception of a uniform space
|  */
| export interface IRowHeightException {
|   /**
|    * reference index
 @ ./src/index.ts 3:0-24 3:0-24

ERROR in ./src/mixin/PrefetchMixin.ts
Module parse failed: Unexpected token (4:7)
You may need an appropriate loader to handle this file type.
| import {EScrollResult, IMixin, IMixinAdapter} from './IMixin';
|
| export interface IPrefetchRendererOptions {
|   /**
|    * number of rows to prefetch
 @ ./src/mixin/index.ts 2:0-57 2:0-57
 @ ./src/index.ts

ERROR in ./src/mixin/IMixin.ts
Module parse failed: Unexpected token (6:7)
You may need an appropriate loader to handle this file type.
|  * scrolling result
|  */
| export enum EScrollResult {
|   /**
|    * nothing has changed
 @ ./src/mixin/index.ts 1:0-75 1:0-75 1:0-75 1:0-75 1:0-75
 @ ./src/index.ts

ERROR in ./src/style/StyleManager.ts
Module parse failed: Unexpected token (7:10)
You may need an appropriate loader to handle this file type.
|  */
| export default class StyleManager {
|   private readonly rules = new Map<string, string>();
|   private readonly node: HTMLStyleElement;
|
 @ ./src/style/index.ts 3:0-55 3:0-55
 @ ./src/index.ts

ERROR in ./src/ARowRenderer.ts
Module parse failed: Unexpected token (8:7)
You may need an appropriate loader to handle this file type.
| import {addScroll, removeScroll, IScrollInfo, IDelayedMode, defaultMode} from './internal';
|
| export declare type IRowRenderContext = IExceptionContext;
|
| export interface IRowRendererOptions {
 @ ./src/index.ts 2:0-31 2:0-31

ERROR in ./src/ACellRenderer.ts
Module parse failed: Unexpected token (9:7)
You may need an appropriate loader to handle this file type.
|
|
| export declare type ICellRenderContext<T extends IColumn> = ICellAdapterRenderContext<T>;
|
| /**
 @ ./src/index.ts 1:0-32 1:0-32

ERROR in ./src/table/MultiTableRowRenderer.ts
Module parse failed: Unexpected token (9:7)
You may need an appropriate loader to handle this file type.
|  * basic interface of a table section
|  */
| export interface ITableSection {
|   readonly id: string;
|   readonly width: number;
 @ ./src/table/index.ts 2:0-8:33 2:0-8:33 2:0-8:33 2:0-8:33 2:0-8:33 2:0-8:33
 @ ./src/index.ts

ERROR in ./src/animation/index.ts
Module parse failed: Unexpected token (9:7)
You may need an appropriate loader to handle this file type.
|  * different row animation modes
|  */
| export enum EAnimationMode {
|   /**
|    * plain update existed both before and after
 @ ./src/index.ts 8:0-28 8:0-28

ERROR in ./src/table/ACellTableSection.ts
Module parse failed: Unexpected token (9:7)
You may need an appropriate loader to handle this file type.
| import {ITableSection} from './MultiTableRowRenderer';
|
| export declare type ICellRenderContext<T extends IColumn> = ICellAdapterRenderContext<T>;
|
| /**
 @ ./src/table/index.ts 1:0-101 1:0-101 1:0-101
 @ ./src/index.ts
```

### Expected behavior

 Compiles without error

**Comment by moritzschaefer** [comment]
Btw these errors come up first after work-arounding this error:
```
ERROR in ./src/index.ts
Module parse failed: Unexpected token (10:8)
You may need an appropriate loader to handle this file type.
| export * from './animation';
|
| declare const __VERSION__: string;
| export const version = __VERSION__;
|

```

**Comment by sgratzl** [comment]
fixed with v1.1.0-alpha.4
